src/sendData.cpp

```diff
@@ -288,11 +290,43 @@ bool dataSendHandler::send_to_brewstatus()
             }
         }
         brewStatusTicker.once(config.brewstatusPushEvery, [](){send_brewStatus = true;}); // Set up subsequent send to Brew Status
+        send_lock = false;
     }
-    send_lock = false;
     return result;
 }
 
+bool dataSendHandler::http_send_json(const char * url, const char * payload)
+{
+    int httpResponseCode;
+    StaticJsonDocument<GF_SIZE> retval;
+    http.setFollowRedirects(HTTPC_FORCE_FOLLOW_REDIRECTS);
+    http.setConnectTimeout(6000);
+    http.setReuse(false);
+    
+    secureClient.setInsecure();
+
+    http.addHeader(F("Content-Type"), F("application/json"));
+    http.addHeader(F("Accept"), F("application/json"));
+    httpResponseCode = http.POST(payload);
+
+    if (httpResponseCode >= 400) {
+        Log.error(F("HTTP error %d: %s, %s.\r\n"), httpResponseCode, http.errorToString(httpResponseCode).c_str(), http.getString().c_str());
+        return false;
+    }
+
+    if (!http.begin(secureClient, url)) {
+        Log.error(F("Unable to create secure connection to %s.\r\n"), url);
+        return false;
+    }
+
+    deserializeJson(retval, http.getString().c_str());
+
+    http.end();
+    retval.clear();
+
+    return true;
+}
```

Comment:
Out of curiosity, why did you decide to break this out? Does the `send_to_url` method not work for Grainfather?

Reply:
It is the first approach to replace send_to_url. It does a lot of stuff there that HTTPClient does already. Especially the string compare with "HTTP/1.1 200" is quite dangerous and doen't work with Grainfather. The Grainfather API uses HTTP/2 with the Status 201. Generally only 400+ status values are errors.

send_to_google uses HTTPClient already.

Reply:
This makes sense - replacing or refactoring `send_to_url` was something I was looking at in the last major rewrite but never got to it.  I thought there might be a better way to toggle between the targets and just sort of forgot to dig further.  

IIRC, `HTTPClient` does become less perfect when we have non-standard ports, so that would need to be figured out before going any further.

Reply:
That makes sense - converting the endpoints we can (Brewers Friend, Brewfather, etc.) to use `http_send_json` or equivalent might be the way to go and just leave `send_to_url` for custom endpoints (which might use mDNS or non-standard ports)